### PR TITLE
Review CUDA and HIP configuration in CMake

### PR DESCRIPTION
**Pull request by pelesh**
Review and if necessary refactor CMake configuration in `hip-first-take-dev` branch to make it ready for merging into `develop`.

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -12,19 +12,19 @@
             "description": "Base config to build with CUDA",
             "binaryDir": "${sourceDir}/build",
             "installDir": "${sourceDir}/install",
-            "generator": "Unix Makefiles"
+            "generator": "Unix Makefiles",
+            "cacheVariables": {
+                "RESOLVE_USE_CUDA": "ON",
+                "RESOLVE_USE_GPU": "ON"
```
Comment (nkoukpaizan) [inline CMakePresets.json:18]:
```suggestion
```

The CMakeLists is now designed to set `RESOLVE_USE_GPU` based on `RESOLVE_USE_CUDA`.

**Review by nkoukpaizan — APPROVED** [review]

File: resolve/CMakeLists.txt [hunk 2]
```diff
@@ -86,17 +61,22 @@ set(ReSolve_Targets_List
...
 if(RESOLVE_USE_HIP)
+  add_subdirectory(hip)
+  target_link_libraries(resolve_tpl INTERFACE resolve_hip)
   set(ReSolve_Targets_List ${ReSolve_Targets_List} resolve_backend_hip)
 endif()
 
 # If no GPU support is enabled, link to dummy device backend
 if(NOT RESOLVE_USE_GPU)
+  add_subdirectory(cpu)
     set(ReSolve_Targets_List ${ReSolve_Targets_List} resolve_backend_cpu)
```
Comment (nkoukpaizan) [inline resolve/CMakeLists.txt:80]:
```suggestion
  set(ReSolve_Targets_List ${ReSolve_Targets_List} resolve_backend_cpu)
```

File: resolve/CMakeLists.txt [hunk 3]
```diff
@@ -37,45 +37,20 @@ set(ReSolve_HEADER_INSTALL
...
 # Next build vector and matrix objects that may use this backend.
 add_subdirectory(vector)
 add_subdirectory(matrix)
 
-
 # Build shared library ReSolve
 add_library(resolve_tpl INTERFACE)
 
 if(RESOLVE_USE_KLU)
     target_link_libraries(resolve_tpl INTERFACE KLU)
```
Comment (nkoukpaizan) [inline resolve/CMakeLists.txt:51]:
```suggestion
  target_link_libraries(resolve_tpl INTERFACE KLU)    
```

File: CMakePresets.json [hunk 4]
```diff
@@ -12,19 +12,18 @@
             "description": "Base config to build with CUDA",
             "binaryDir": "${sourceDir}/build",
             "installDir": "${sourceDir}/install",
-            "generator": "Unix Makefiles"
+            "generator": "Unix Makefiles",
+            "cacheVariables": {
+                "RESOLVE_USE_CUDA": "ON",
+            }
```
Comment (pelesh) [inline CMakePresets.json:18]:
```suggestion
                "RESOLVE_USE_CUDA": "ON"
            }
```

**Pull request merged**